### PR TITLE
Implement external switch loop protection (standalone and stacking).

### PR DESCRIPTION
**Pull request by anarkiwi**
Also fixes non hardware ports 1-4 tests.

### CODE REVIEW

File: faucet/config_parser_util.py [hunk 1]
```diff
@@ -184,8 +184,7 @@ def config_changed(top_config_file, new_top_config_file, config_hashes):
         if config_hash and not config_file_exists:
             return True
         # Config file hash has changed = reload.
-        if config_file_exists:
-            new_config_hash = config_file_hash(config_file)
-            if new_config_hash != config_hash:
-                return True
+        new_config_hash = config_file_hash(config_file)
```
Comment (gizmoguy) [inline faucet/config_parser_util.py]: This change erroneously reverts #2702

Comment (anarkiwi) [reply faucet/config_parser_util.py]: Fixed.

**Comment by gizmoguy** [comment]
Merging this commit will likely remove the attribution to @grafnu. Can we do a rebase instead.

**Comment by anarkiwi** [comment]
Rebase messed up, regenerating.

**Pull request closed without merging**